wfd/public/media_manager.h

```diff
@@ -172,11 +166,34 @@ class WFD_EXPORT SourceMediaManager : public MediaManager {
    * @param list of H264 formats that are supported by remote device
    * @return optimal H264 video format
    */
-  virtual SelectableH264VideoFormat FindOptimalFormat(
+  virtual SelectableH264VideoFormat FindOptimalVideoFormat(
       const NativeVideoFormat& remote_device_native_format,
       const std::vector<SelectableH264VideoFormat>& remotely_supported_formats) const;
 
   /**
+   * Gets optimal H264 format @see SetOptimalVideoFormat
+   *
+   * @return optimal H264 format
+   */
+  virtual SelectableH264VideoFormat GetOptimalVideoFormat() const = 0;
+
+  /**
+   * Initializes optimal audio codec
+   * The optimal audio codec will be returned by GetOptimalAudioFormat
+   *
+   * @param sink_supported_codecs list of the codecs supported by sink
+   * @return true if optimal codec can be found among the given list items, false otherwise
+   */
+  virtual bool InitOptimalAudioFormat(const std::vector<AudioCodec>& sink_supported_codecs) = 0;
+
```

Comment:
video format cap negotiations API needs fixing, I'll update it with following patches.